<launch>
    <param name="/use_sim_time" value="true"/>

    <arg name="stereo" default="camera"/>
    <arg name="elas_ns" default="elas"/>
    
    <node name="camera_info_publisher" pkg="my_scripts" type="yaml_to_camerainfo.py" output="screen">
        <param name="left_camera_yaml" value="/root/catkin_ws/data/Euroc/left_cam.yaml"/>
        <param name="right_camera_yaml" value="/root/catkin_ws/data/Euroc/right_cam.yaml"/>
    </node>
    
    <node name="$(arg elas_ns)" pkg="elas_ros" type="elas_ros" output="screen">
        <remap from="stereo/left/image"        to="/cam0/image_raw"/>
        <remap from="stereo/left/camera_info"  to="/camera/left/camera_info"/>
        <remap from="stereo/right/image"       to="/cam1/image_raw"/>
        <remap from="stereo/right/camera_info" to="/camera/right/camera_info"/>

	<param name="point_cloud_frame_id" value="stereo_camera_frame"/>
        <param name="disp_min" type="int" value="0"/>
        <param name="disp_max" type="int" value="255"/>
        <param name="support_threshold" type="double" value="0.55"/>
        <param name="support_texture" type="int" value="10"/>
        <param name="candidate_stepsize" type="int" value="5"/>
        <param name="incon_window_size" type="int" value="5"/>
        <param name="incon_threshold" type="int" value="5"/>
        <param name="incon_min_support" type="int" value="5"/>
        <param name="add_corners" type="bool" value="0"/>
        <param name="grid_size" type="int" value="20"/>
        <param name="beta" type="double" value="0.02"/>
        <param name="gamma" type="double" value="3"/>
        <param name="sigma" type="double" value="1"/>
        <param name="sradius" type="double" value="2"/>
        <param name="match_texture" type="int" value="1"/>
        <param name="lr_threshold" type="int" value="1"/>
        <param name="speckle_sim_threshold" type="double" value="1"/>
        <param name="speckle_size" type="int" value="50"/>
        <param name="ipol_gap_width" type="int" value="20"/>
        <param name="filter_median" type="bool" value="0"/>
        <param name="filter_adaptive_mean" type="bool" value="1"/>
        <param name="postprocess_only_left" type="bool" value="1"/>
        <param name="subsampling" type="bool" value="0"/>

        <param name="approximate_sync" value="true" />
        <param name="queue_size" type="int" value="5"/>
        </node>
 
    <node pkg="tf" type="static_transform_publisher" name="body_to_stereo_broadcaster"
          args="0.0148655429818 0.0216401454975 0.00283683617911 -1.5707963 0 -1.5707963 body stereo_camera 100" />
          
          </launch>
